src/core/thread/mle.cpp

```diff
@@ -1416,6 +1416,25 @@ otError Mle::AppendAddressRegistration(Message &aMessage, AddressRegistrationMod
     VerifyOrExit(aMode != kAppendMeshLocalOnly, OT_NOOP);
     counter++;
 
+#if OPENTHREAD_CONFIG_TMF_DUA_REGISTRATION_ENABLE
+    if (Get<BackboneRouter::Leader>().HasDomainPrefix())
+    {
+        const Ip6::Address &address = GetDomainUnicastAddress();
+
+        if (Get<ThreadNetif>().IsUnicastAddress(address))
+        {
+            OT_ASSERT(Get<NetworkData::Leader>().GetContext(address, context) == OT_ERROR_NONE);
```

Comment:
Should not put real logic code in `OT_ASSERT` because `OT_ASSERT` can be turned off.

Reply:
separated `OT_ASSERT` out.